File: 940-fruit-into-baskets/fruit-into-baskets.py
class Solution:
    def totalFruit(self, frs: List[int]) -> int:
        l,r,res=0,0,0
        s={}
        while r<len(frs):
            if frs[r] not in s:
                s[frs[r]]=1
            else:
                s[frs[r]]+=1
            if len(s)>2:
                s[frs[l]]-=1
                if s[frs[l]]==0:
                    del s[frs[l]]
                l+=1
            res=max(res,r-l+1)
            r+=1
        print(s)
        return res
        

File: 940-fruit-into-baskets/fruit-into-baskets.cpp
class Solution {
public:
    int totalFruit(vector<int>& fruits) {
        int l=0,r=0,res=0;
        unordered_map<int,int>mpp;
        while (r< fruits.size()){
            mpp[fruits[r]]++;

            if(mpp.size()>2){
                mpp[fruits[l]]--;
                if (mpp[fruits[l]]==0){
                    mpp.erase(fruits[l]);
                }
                l++;
            }

            res=max(res,r-l+1);
            r++;
        }
        return res;
    }
};